eng/pipelines/templates/jobs/ci.yml

```diff
@@ -168,14 +168,16 @@ jobs:
           workingDirectory: '$(Agent.BuildDirectory)'
       - task: PowerShell@2
         displayName: "Verify generated code"
+        env:
```

Comment:
Does this just save time is that the only reason you are disabling it here?

Reply:
Sparse checkout saves time but the build fails on sparse git repositories because SourceLink doesn't support them yet.  So we have to disable sourcebuild when enabling sparse checkout.

Few more details: https://github.com/Azure/azure-sdk-for-net/issues/26166